src/core/agent/services/identifierService.ts

```diff
@@ -375,6 +377,86 @@ class IdentifierService extends AgentService {
     });
     return multisigId;
   }
+
+  async rotateMultisig(ourIdentifier: string): Promise<string | undefined> {
+    const metadata = await this.getMetadataById(ourIdentifier);
+    if (metadata.method !== IdentifierType.KERI) {
+      throw new Error(IdentifierService.ONLY_CREATE_ROTATION_WITH_AID);
+    }
+    if (!metadata.signifyName) {
+      throw new Error(IdentifierService.AID_MISSING_SIGNIFY_NAME);
+    }
+    // Make sure no non-Keri contacts get passed into this function
+    const members = await this.agent.modules.signify.getMultisigMembers(
+      metadata.signifyName
+    );
+    const multisigMembers: Aid[] = [];
+    await Promise.all(
+      members?.signing?.map(async (signing: any) => {
+        const aid = await this.agent.modules.signify.queryKeyState(
+          signing.aid,
+          "0"
+        );
+        multisigMembers.push({ state: aid.response } as Aid);
+      })
+    );
```

Comment:
I would add a check here that `multisigMembers` was of the correct number (for now, multi-sig of 3 will rotate to multi-sig of 3) - just in case any cannot be found anymore

Reply:
Hmm, it mean when we cannot find the AID with member, we will stop the rotation or join rotation, right?

Reply:
Just for the rotate (not join) - but yes, if we cannot find one of the originally members, we need to stop.

In the future, we will support more complex rotations where you change the amount of people participating or rotating but that is quite complex so for now just throw an error if we cannot find all members.